core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -309,65 +304,20 @@ protected enum CommitStatus {
-    int maxAttempts =
-        PropertyUtil.propertyAsInt(
-            config.properties(), COMMIT_NUM_STATUS_CHECKS, COMMIT_NUM_STATUS_CHECKS_DEFAULT);
-    long minWaitMs =
-        PropertyUtil.propertyAsLong(
-            config.properties(),
-            COMMIT_STATUS_CHECKS_MIN_WAIT_MS,
-            COMMIT_STATUS_CHECKS_MIN_WAIT_MS_DEFAULT);
-    long maxWaitMs =
-        PropertyUtil.propertyAsLong(
-            config.properties(),
-            COMMIT_STATUS_CHECKS_MAX_WAIT_MS,
-            COMMIT_STATUS_CHECKS_MAX_WAIT_MS_DEFAULT);
-    long totalRetryMs =
-        PropertyUtil.propertyAsLong(
-            config.properties(),
-            COMMIT_STATUS_CHECKS_TOTAL_WAIT_MS,
-            COMMIT_STATUS_CHECKS_TOTAL_WAIT_MS_DEFAULT);
-
-    AtomicReference<CommitStatus> status = new AtomicReference<>(CommitStatus.UNKNOWN);
-
-    Tasks.foreach(newMetadataLocation)
-        .retry(maxAttempts)
-        .suppressFailureWhenFinished()
-        .exponentialBackoff(minWaitMs, maxWaitMs, totalRetryMs, 2.0)
-        .onFailure(
-            (location, checkException) ->
-                LOG.error("Cannot check if commit to {} exists.", tableName(), checkException))
-        .run(
-            location -> {
-              TableMetadata metadata = refresh();
-              String currentMetadataFileLocation = metadata.metadataFileLocation();
-              boolean commitSuccess =
-                  currentMetadataFileLocation.equals(newMetadataLocation)
-                      || metadata.previousFiles().stream()
-                          .anyMatch(log -> log.file().equals(newMetadataLocation));
-              if (commitSuccess) {
-                LOG.info(
-                    "Commit status check: Commit to {} of {} succeeded",
-                    tableName(),
-                    newMetadataLocation);
-                status.set(CommitStatus.SUCCESS);
-              } else {
-                LOG.warn(
-                    "Commit status check: Commit to {} of {} unknown, new metadata location is not current "
-                        + "or in history",
-                    tableName(),
-                    newMetadataLocation);
-              }
-            });
-
-    if (status.get() == CommitStatus.UNKNOWN) {
-      LOG.error(
-          "Cannot determine commit state to {}. Failed during checking {} times. "
-              + "Treating commit state as unknown.",
-          tableName(),
-          maxAttempts);
-    }
-    return status.get();
+    return checkCommitStatus(
+        tableName(), newMetadataLocation, config.properties(), this::loadMetadataLocations);
+  }
+
+  protected List<String> loadMetadataLocations() {
+    TableMetadata metadata = refresh();
+    ImmutableList.Builder<String> builder = ImmutableList.builder();
+    return builder
+        .add(metadata.metadataFileLocation())
+        .addAll(
+            metadata.previousFiles().stream()
```

Comment:
Should we create an issue for keeping historical files for view? 
Since current approach is nowhere keeping the previous files. 
```
public interface ViewHistoryEntry {
  /** Return the timestamp in milliseconds of the change */
  long timestampMillis();

  /** Return ID of the new current version */
  int versionId();
}
```

Reply:
I'm not sure if we really want to do this just for Hive. Is there maybe a better way to solve this for view?

Reply:
There could be cases when we have a HMS commit failure, but the actual commit is successful, we just don't receive the answer. We try to check the current history for our commit. If we find it, then we say that the commit was successful independently of the previous HMS response

Reply:
@pvary / @szehon-ho what are we proposing here? Since [ViewHistoryEntry](https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/view/ViewHistoryEntry.java) doesn't have previous metadata files information . So we can't compare with metadata location. 
Also I am curious what will be harm if we match `versionId` for View and `snapshotId` for table instead of `metadataLocation` ?

Reply:
I have used now `metadataLocation` check with table and `versionId` for View. 
Table -
```
checkCommitStatus(
        fullName,
        newMetadata.metadataFileLocation(),
        newMetadata.properties(),
        () -> checkNewVersion(((ViewMetadata) newMetadata).currentVersionId()));
```
View -
```
checkCommitStatus(
        fullName,
        newMetadata.metadataFileLocation(),
        newMetadata.properties(),
        () -> checkNewVersion(((ViewMetadata) newMetadata).currentVersionId()));
```

Reply:
HI, I checked with @RussellSpitzer who authored this originally for Hive.  It's a real issue for Hive tables, because frequent commits lead to situations where we lose contact with HMS or HMS goes down but the commit goes through.  Then it takes awhile to reconnect, and in the meantime there's other commits.  It seems it was frequent for Hive tables, especially where we commit quite often (streaming for example).

let's revert your latest change then to check using version.  It is not correct, version is not supposed to be incremented for every update for view in my understanding, just when the view representation changes?

Reply:
If that's the case , Should we do commitStatusValidation ? It seems for View it doesn't need following kind of [checks](https://github.com/apache/iceberg/blob/main/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L285-L292) 
```
commitStatus = checkCommitStatus(newMetadataLocation, metadata);
        switch (commitStatus) {
          case SUCCESS:
            break;
          case FAILURE:
            throw e;
          case UNKNOWN:
            throw new CommitStateUnknownException(e);
```
For view should we `throw new CommitStateUnknownException(e);`  in case of error?

Reply:
Yes, we need these @nk1506 , the logic should be exactly the same, except for the comparison of metadata in checkCommitStatus.  In HiveTableOperations we compare against all metadata paths in the history, in HiveViewOperations we compare only the expected one.